runtime/Swift/Sources/Antlr4/misc/MurmurHash.swift

```diff
@@ -111,14 +112,55 @@ public final class MurmurHash {
     /// - Returns: the hash code of the data
     /// 
     public static func hashCode<T:Hashable>(_ data: [T], _ seed: Int) -> Int {
-        var hash: Int = initialize(seed)
-        for value: T in data {
-            //var hashValue = value != nil ?  value.hashValue : 0
-            hash = update(hash, value.hashValue)
+        var hash = initialize(UInt32(truncatingIfNeeded: seed))
+        for value in data {
+            hash = update(hash, value)
         }
 
-        hash = finish(hash, data.count)
-        return hash
+        return finish(hash, data.count)
+    }
+
+    ///
+    /// Compute a hash for the given String and seed.  The String is encoded
+    /// using UTF-8, then the bytes are interpreted as unsigned 32-bit
+    /// little-endian values, giving UInt32 values for the update call.
+    ///
+    /// If the bytes do not evenly divide by 4, the final bytes are treated
+    /// slightly differently (not doing the final rotate / multiply / add).
+    ///
+    /// This matches the treatment of byte sequences in publicly available
+    /// test patterns (see MurmurHashTests.swift) and the example code on
+    /// Wikipedia.
+    ///
+    public static func hashString(_ s: String, _ seed: UInt32) -> UInt32 {
```

Comment:
I don't see where this function is used?

Reply:
It's only used by the unit tests.  The test patterns that I found were built around UTF-8 strings as input.  The UTF-8 strings are then decoded to UInt32 arrays before being passed to the hash function.  (See `MurmurHashTests.testMurmurHash`.)

In our codebase proper, we're only using MurmurHash with UInt32 (or lists of them) as input.  We don't have anywhere where the input is passed as a UTF-8 string.  I added hashString so that we can check that our MurmurHash matches the public definition of MurmurHash, but other than for that, we're not using it.

Reply:
I see. We need it to verify the correctness of the implementation.